src/ruby/puppetserver-lib/puppet/server/ast_compiler.rb

```diff
@@ -24,26 +24,79 @@ def self.compile(compile_options)
       end
 
       def self.compile_ast(code, compile_options)
+        # TODO: move requires to optimal place and guard against FOSS not having bolt lib code
+        # CODEREVIEW: where should these requires be, and what should happen if they are not found?
+        require 'bolt'
+        require 'bolt/target'
+        require 'bolt/apply_inventory'
+        require 'bolt/apply_target'
+        require 'bolt/apply_result'
+        require 'addressable'
+
+        Puppet[:rich_data] = true
         Puppet[:node_name_value] = compile_options['certname']
```

Comment:
Did we miss this when reviewing our threadsafety work? I think the implementation we landed on makes this safe, but I'm pretty sure only by an accident of the implementation.

Reply:
I dont have context here on this question. I'm guessing its a question to Froyo?

Reply:
Yeah, sorry that was for Froyo